indexing-service/src/main/java/org/apache/druid/indexing/worker/IntermediaryDataManager.java

```diff
@@ -336,6 +337,7 @@ long addSegment(String supervisorTaskId, String subTaskId, DataSegment segment,
   @Nullable
   public File findPartitionFile(String supervisorTaskId, String subTaskId, Interval interval, int partitionId)
   {
+    TaskIdUtils.validateId("supervisorTaskId", supervisorTaskId);
```

Comment:
What do you think about moving the validation up the stack to right when the external data enters druid (i.e., in the API resource layer like `ShuffleResource.getPartition()/deletePartitions()`)? For example, in the future, there may be clients of the lower level code that do not need to do the validation.

Reply:
It seems safer to me to be here where the problem is possible so that any future potential caller wouldn't need to worry about forgetting to do this and avoid the possibility.